test/unit/native_deps/native_deps_test.bzl

```diff
@@ -243,13 +254,56 @@ def _native_dep_test():
         target_under_test = ":cdylib_has_native_dep_and_alwayslink",
     )
 
+def _linkopts_propagate_test_impl(ctx):
+    env = analysistest.begin(ctx)
+    tut = analysistest.target_under_test(env)
+    actions = analysistest.target_actions(env)
+    action = actions[0]
+
+    # Ensure linkopts from direct (-Llinkoptdep1) and transitive
+    # (-Llinkoptdep2) dependencies are propagated.
+    linkopt_1_found = False
```

Comment:
Could you also assert the order in which linkopts appear? You maybe can use test/unit/common.bzl `assert_list_contains_adjacent_elements`?